src/cockpit/channel.py

```diff
@@ -394,3 +394,29 @@ def do_data(self, data):
             data = bytes(data)
 
         self.receive_queue.put_nowait(data)
+
+
+class GeneratorChannel(Channel):
+    """A trivial Channel subclass for sending data from a generator with flow control.
+
+    Calls the .do_yield_data() generator with the options from the open message
+    and sends the data which it yields.  If the generator returns a value it
+    will be used for the close message.
+    """
+    DataGenerator = Generator[bytes, None, Optional[Dict[str, object]]]
+    __generator: DataGenerator
+
+    def do_yield_data(self, size_hint: int, options: dict[str, object]) -> DataGenerator:
```

Comment:
Do you actually plan to ever send something else than `Channel.CHANNEL_FLOW_WINDOW` as `size_hint`? If not, let's just drop this? When reading the implementation, I still keep asking myself "what kind of values do I need to handle for size_hint"?

Reply:
not `WINDOW`, but `PING`.  I think these constants are long overdue for renaming.

Reply:
I also just realized that the way `fslist1` is implemented is bonkers.  We're gonna send a ping after every single directory entry, which seems excessive.  This needs a bit of thinking.

Reply:
The code really [calls it with FLOW_WINDOW](https://github.com/cockpit-project/cockpit/pull/18257/files#diff-50b095c0e61dea51dd6279554f093ec49005bb91f2e2078acc74a4d58eeba9b7R413), so I suppose you want to change tihs?

Do you just want to drop the fslist1 patch for now?

Reply:
Yes.

I have some vague plan in the future to support merging messages and supporting `str` and json as well.